File: easy-problems/src/main/scala/com/leetcode/problems/easy/ListNode.scala
package com.leetcode.problems.easy

class ListNode(_x: Int = 0, _next: ListNode = null) {
  var next: ListNode = _next
  var x: Int = _x

  override def toString: String = s"(${this.x})"
}

object ListNode {
  def apply(x: Int): ListNode = new ListNode(x)

  def apply(x: Int, next: ListNode): ListNode = new ListNode(x, next)
}

File: medium-problems/src/main/java/com/leetcode/problems/medium/ListNode.java
package com.leetcode.problems.medium;

final class ListNode {
    int val;
    ListNode next;

    ListNode(int val) {
        this.val = val;
    }

    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
    @Override
    public String toString() {
        return "ListNode(" + val + ")";
    }
}
